OpenRA.Mods.Common/Traits/Render/WithDamageOverlay.cs

```diff
@@ -16,7 +16,7 @@
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Renders an overlay when the actor is taking heavy damage.")]
-	public class WithDamageOverlayInfo : TraitInfo, Requires<RenderSpritesInfo>
+	public class WithDamageOverlayInfo : TraitInfo, Requires<RenderSpritesInfo>, NotBefore<BodyOrientationInfo>, IRulesetLoaded
```

Comment:
the proper way we check is by delaying the acquisition of the trait to INotifyCreated

Reply:
Or using `Requires`? That way the check in `RulesetLoaded` is redundant.